service-mesh/config.go

```diff
@@ -397,3 +412,129 @@ func downstreamTLSTransportSocket(requireClientCertificate bool) (*envoyCoreV3.T
 		},
 	}, nil
 }
+
+func addBlackHoleToConfig(config *envoyConfigBootstrapV3.Bootstrap) error {
+	// Add blackHoleCluster
+	config.StaticResources.Clusters = append(config.StaticResources.Clusters, blackHoleCluster())
+
+	// Add BlackHole matching to all listeners
+	for _, listener := range config.StaticResources.Listeners {
+		if listener.FilterChainMatcher != nil {
+			return fmt.Errorf("listener %s already has a filterChainMatcher", listener.Name)
+		}
+		listenPort := listener.Address.GetSocketAddress().GetPortValue()
+		if listenPort == 0 {
+			return fmt.Errorf("listener %s listens on port 0", listener.Name)
+		}
+		if listenPort != 15006 && listenPort != 15007 {
```

Comment:
Can we make these arguments and use constants where we're calling `addBlackHoleToConfig`? That'd allow easier customization later on - I've been wondering lately when we'll have the first user that wants these ports for their app :)